pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -489,3 +490,95 @@ func (f *ring1Factory) OpenAPISchema(cacheDir string) (openapi.Resources, error)
 	// Delegate to the OpenAPIGetter
 	return f.openAPIGetter.getter.Get()
 }
+
+func (f *ring1Factory) ValidResourcesFromDiscoveryClient() string {
```

Comment:
I think this method ought to return `[]string` of just the resources strings

Reply:
resources strings only contain resource name ? or we can directly return `[]APIResources` then write a wrapper/helper function to concatenate those `APIResource.Name` and `APIREsource.ShortNames` to string output ?